File: src/main/java/ru/vito/desktop/app/Runner.java
package ru.vito.desktop.app;

import ru.vito.desktop.app.models.DefinedUsers;
import ru.vito.desktop.app.service.impl.Definer;
import ru.vito.desktop.app.service.impl.Parser;
import ru.vito.desktop.app.service.impl.Reader;

import java.util.HashSet;
import java.util.Map;

public class Runner {

    /**
     * @param strings receiving from input string array
     */
    public static void main(final String[] strings) {

        System.out.println(run(strings));
    }

    private static String run(final String[] strings) {

        final Map<String, HashSet<String>> userToEmailsMap =
            new Parser().parse(
                new Reader().read(strings)
            );

        final DefinedUsers definedUsers =
            new Definer()
                .define(userToEmailsMap);

        return definedUsers.toString();
    }
}

File: src/main/kotlin/ru/vito/desktop/app/Runner.kt
package ru.vito.desktop.app

import ru.vito.desktop.app.service.impl.Definer
import ru.vito.desktop.app.service.impl.Parser
import ru.vito.desktop.app.service.impl.Reader
import java.util.*

/**
 * @param strings receiving from input string array
 */
fun main(strings: Array<String>) {

    println(Runner.run(strings))
}

class Runner {
    companion object {
        fun run(strings: Array<String>): String {

            val userToEmailsMap: Map<String, HashSet<String>> = Parser().parse(
                    Reader().read(strings)
            )

            val definedUsers = Definer()
                    .define(userToEmailsMap)

            return definedUsers.toString()
        }
    }
}